aptos-move/aptos-vm/src/block_executor/vm_wrapper.rs

```diff
@@ -69,16 +68,15 @@ impl<'a, S: 'a + StateView + Sync> ExecutorTask for AptosExecutorTask<'a, S> {
             .vm
             .execute_single_transaction(txn, &view.as_move_resolver(), &log_context)
         {
-            Ok((vm_status, mut output_ext, sender)) => {
+            Ok((vm_status, mut vm_output, sender)) => {
                 if materialize_deltas {
-                    // Keep TransactionOutputExt type for wrapper.
-                    output_ext = TransactionOutputExt::new(
-                        DeltaChangeSet::empty(),                  // Cleared deltas.
-                        output_ext.into_transaction_output(view), // Materialize.
-                    );
+                    // TODO: Integrate delta application failure.
+                    vm_output = vm_output
+                        .try_materialize(view)
+                        .expect("Delta materialization failed");
```

Comment:
@vusirikala In this PR I am shifting the error up to here, with our work on getting rid of `materialize_deltas` should be easy to adjust?